### PR TITLE
init: Some small chainstate load improvements

### PR DESCRIPTION
**Pull request by sedited**
These are mostly followups from #30968, making the code, log lines, error messages, and comments more consistent.

The last commit is an attempt at improving the error reporting when loading the chainstate. It aims to more cleanly distinguish between errors arising from a specific database, and errors where the culprit may be less clear.

### CODE REVIEW

File: src/node/chainstate.cpp [hunk 1]
```diff
@@ -41,12 +41,16 @@ static ChainstateLoadResult CompleteChainstateInitialization(
...
-        .wipe_data = options.wipe_block_tree_db,
-        .options = chainman.m_options.block_tree_db});
+    try {
+        pblocktree = std::make_unique<BlockTreeDB>(DBParams{
+            .path = chainman.m_options.datadir / "blocks" / "index",
+            .cache_bytes = static_cast<size_t>(cache_sizes.block_tree_db),
+            .memory_only = options.block_tree_db_in_memory,
+            .wipe_data = options.wipe_block_tree_db,
+            .options = chainman.m_options.block_tree_db});
+    } catch (dbwrapper_error&) {
```
Comment (stickies-v) [inline src/node/chainstate.cpp:51]: Note: instantiating a `CDBWrapper` can throw other errors such as a `fs::filesystem_error` from [here](https://github.com/bitcoin/bitcoin/blob/62e4516722115c2d5aeb6c197abc73ca7c078b23/src/dbwrapper.cpp#L240), but I suppose that is not something the user can recover from with a reindex so not catching that here seems correct.

I think it logging the `dbwrapper_error` would be useful though?

<details>
<summary>git diff on 1bec418e77</summary>

```diff
diff --git a/src/node/chainstate.cpp b/src/node/chainstate.cpp
index 76094105fa..b9c3981910 100644
--- a/src/node/chainstate.cpp
+++ b/src/node/chainstate.cpp
@@ -48,8 +48,10 @@ static ChainstateLoadResult CompleteChainstateInitialization(
             .memory_only = options.block_tree_db_in_memory,
             .wipe_data = options.wipe_block_tree_db,
             .options = chainman.m_options.block_tree_db});
-    } catch (dbwrapper_error&) {
-        return {ChainstateLoadStatus::FAILURE, _("Error opening block database")};
+    } catch (dbwrapper_error& err) {
+        auto msg{_("Error opening block database")};
+        LogError("%s: %s\n", msg.original, err.what());
+        return {ChainstateLoadStatus::FAILURE, msg};
     }
 
     if (options.wipe_block_tree_db) {
@@ -116,8 +118,10 @@ static ChainstateLoadResult CompleteChainstateInitialization(
                 /*cache_size_bytes=*/chainman.m_total_coinsdb_cache * init_cache_fraction,
                 /*in_memory=*/options.coins_db_in_memory,
                 /*should_wipe=*/options.wipe_chainstate_db);
-        } catch (dbwrapper_error&) {
-            return {ChainstateLoadStatus::FAILURE, _("Error opening coins database")};
+        } catch (dbwrapper_error& err) {
+            auto msg{_("Error opening coins database")};
+            LogError("%s: %s\n", msg.original, err.what());
+            return {ChainstateLoadStatus::FAILURE, msg};
         }
 
         if (options.coins_error_cb) {

```
</details>

**Review by stickies-v — COMMENTED** [review]
Approach ACK 1bec418e77482a4c53c8f1e7c89a151fdeae0f8e

**Review by mzumsande — COMMENTED** [review]
one more small thing that I noticed reviewing #30968 (it wasn't really on-topic there, but fits with the last commit here): I don't think that we should return `ChainstateLoadStatus::FAILURE` in https://github.com/bitcoin/bitcoin/blob/5837e3463fe188a65d67f96e84cbcca674d61d9e/src/node/chainstate.cpp#L239 (snapshot validation failure). This is not a problem where a general reindex would typically help, and is handled inside of validation (`InvalidateCoinsDBOnDisk`) anyway, so something like `node::ChainstateLoadStatus::FAILURE_FATAL` might be bettter.

**Comment by TheCharlatan** [comment]
Thank you for the reviews @mzumsande and @stickies-v,

Updated 1bec418e77482a4c53c8f1e7c89a151fdeae0f8e -> a7a02b09fafe9d03e5db1fb9d9600f6a13aaa852 ([chainstate_init_followup_0](https://github.com/TheCharlatan/bitcoin/tree/chainstate_init_followup_0) -> [chainstate_init_followup_1](https://github.com/TheCharlatan/bitcoin/tree/chainstate_init_followup_1), [compare](https://github.com/TheCharlatan/bitcoin/compare/chainstate_init_followup_0..chainstate_init_followup_1))

* Took @stickies-v's [suggestion](https://github.com/bitcoin/bitcoin/pull/31046#discussion_r1791672987), implementing additional logging.
* Took @mzumsande's [comment](https://github.com/bitcoin/bitcoin/pull/31046#pullrequestreview-2355081461), added a commit changing a failure to validated a snapshot to fatal.

**Review by mzumsande — COMMENTED** [review]
Code Review / lightly tested ACK 31cc5006c3de4dd6a1f7a238684163956604df45

**Review by BrandonOdiwuor — COMMENTED** [review]
Code Review ACK 31cc5006c3de4dd6a1f7a238684163956604df45. 

Nice work on the chainstate loading improvements, especially the enhanced handling of database-specific errors.

**Review by stickies-v — APPROVED** [review]
ACK 31cc5006c3de4dd6a1f7a238684163956604df45

File: src/node/chainstate.cpp [hunk 2]
```diff
@@ -236,7 +246,7 @@ ChainstateLoadResult LoadChainstate(ChainstateManager& chainman, const CacheSize
             return {init_status, init_error};
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/node/chainstate.cpp:246]: I think your suggestion is a bit better. I'll push it if something else comes up. Looks to me like that code iterated over the chainstates in a previous un-pushed version, that would explain why only the errors are surfaced here. There is also ryanofsky's assumeutxo state cleanup PR that could catch this later if we don't get it here

**Comment by achow101** [comment]
ACK 31cc5006c3de4dd6a1f7a238684163956604df45

**Pull request merged**